components/suggestion/search_channel_suggestion/index.ts

```diff
@@ -2,16 +2,21 @@
 // See LICENSE.txt for license information.
 import {connect} from 'react-redux';
 
-import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
-
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+import {Channel} from 'mattermost-redux/types/channels';
 import {getDirectTeammate} from 'utils/utils';
+import {GlobalState} from 'types/store';
 
 import SearchChannelSuggestion from './search_channel_suggestion';
 
-const mapStateToProps = (state, ownProps) => {
+type OwnProps = {
+    item: Channel;
+}
+
+const mapStateToProps = (state: GlobalState, ownProps: OwnProps) => {
     return {
         teammate: getDirectTeammate(state, ownProps.item.id),
-        currentUser: getCurrentUserId(state),
+        currentUser: getCurrentUser(state),
```

Comment:
Hi @devinbinnie , I change it from "getCurrentUserId" to "getCurrentUserId"  because previous jsx code using getCurrentUserId

Reply:
Is this something we need to change?

Reply:
Hi @devinbinnie , I change it so the currentUser data type will be UserProfile (previously string).

By changing the data type of currentUser to be "UserProfile",  in file "components/suggestion/search_channel_suggestion/search_channel_suggestion.tsx", I can render html tag Avatar to become like this : 
`<Avatar
          username={currentUser.username}
          url={imageURLForUser(getUserIdFromChannelName(currentUser.id, item.name), currentUser.last_picture_update)}
          size='sm'
 />`

Previous code in jsx file like this : 
`<Avatar
          username={item.username}
          url={imageURLForUser(getUserIdFromChannelName(currentUser, item.name), item.last_picture_update)}
          size='sm'
 />`

As you can see in previous code, `item.username` and `item.last_picture_update` will not working if we think item as "Channel" since username and last_picture_update are UserProfile properties.